util/types/field_type.go

```diff
@@ -91,7 +91,7 @@ func AggTypeClass(tps []*FieldType, flag *uint) TypeClass {
 			continue
 		}
 		argTypeClass := argFieldType.ToClass()
-		if argTypeClass == ClassString && mysql.HasBinaryFlag(argFieldType.Flag) {
+		if (IsTypeBlob(argFieldType.Tp) || IsTypeVarchar(argFieldType.Tp) || IsTypeChar(argFieldType.Tp)) && mysql.HasBinaryFlag(argFieldType.Flag) {
```

Comment:
@XuHuaiyu  Since `charset.CollationBin` always come with `mysql.BinaryFlag`, I think we had better modify them together. `types.AggFieldType` may set`mysql.BinaryFlag` for `FieldType.Flag`, but not change `FieldType.Charset` to `charset.CollationBin`. It's a little confusing.